### PR TITLE
Update ethpm uri parser to support default chain ids

### PR DESCRIPTION
**Pull request by njgheorghita**
Updated the ethpm uri parser to...
- support default chain ids of `1`
- support the `ethpm` scheme

### CODE REVIEW

**Comment by njgheorghita** [comment]
ping @marcgarreau whenever you get some time for :eyes: - don't hold back, I like it rough

File: src/utils/uri.ts [hunk 1]
```diff
@@ -38,30 +35,32 @@ export class Erc1319URI {
...
-      return ['', ''];
+      return ['', '', ''];
     }
   }
 
   static parseChainId(url: URL) {
     // todo: support multiple chain ids
     const chainId = +url.port;
-    if (chainId !== 1) {
+    if (chainId === 1 || chainId === 0) {
```
Comment (marcgarreau) [inline src/utils/uri.ts]: Is `url.port` guaranteed to exist? Just a warning that in JS, `+undefined` returns `NaN`, not zero.

File: src/utils/uri.ts [hunk 2]
```diff
@@ -38,30 +35,32 @@ export class Erc1319URI {
           `Invalid package ID: ${packageId}. URI must include both a package name and version.`,
         );
       } else {
-        return packageId;
+        let [packageName, version] = packageId
+        let namespacedAsset = pathElements.splice(1).join("/")
```
Comment (marcgarreau) [inline src/utils/uri.ts]: Worth having a dedicated function to derive the `namespacedAsset` (which can be tested for edge cases)?

File: src/utils/test/uri.ts [hunk 3]
```diff
@@ -1,50 +1,90 @@
-import { Erc1319URI } from 'ethpm/utils/uri';
+import { EthpmURI } from 'ethpm/utils/uri';
+import { URL } from 'url';
 
 describe('validates URIs', () => {
   const validRegistryUris = [
+    'erc1319://snakecharmers.eth',
     'erc1319://snakecharmers.eth:1',
     'erc1319://snakecharmers.eth:1/',
+    new URL('erc1319://snakecharmers.eth:1/'),
```
Comment (marcgarreau) [inline src/utils/test/uri.ts]: Are `url` objects only used internally? Are users expected to/given the option to use them? Just curious/context-gathering, but will note that while flexibility can be nice, adoption can be aided by presenting one clear, "best" way to do the thing.

File: src/utils/uri.ts (hunk 2 above)
Comment (njgheorghita) [reply src/utils/uri.ts]: Hmmm, definitely down the road - but for the current context the `namespacedAsset` is a pretty superfluous datapoint

**Comment by njgheorghita** [comment]
danke @marcgarreau !
